test/controllers/organization_controller_test.exs

```diff
@@ -1,202 +1,97 @@
 defmodule CodeCorps.OrganizationControllerTest do
-  use CodeCorps.ApiCase
-
-  alias CodeCorps.Organization
-  alias CodeCorps.Repo
-  alias CodeCorps.SluggedRoute
+  use CodeCorps.ApiCase, resource_name: :organization
 
   @valid_attrs %{description: "Build a better future.", name: "Code Corps"}
   @invalid_attrs %{name: ""}
 
-  defp build_payload, do: %{ "data" => %{"type" => "organization"}}
-
   describe "index" do
     test "lists all entries on index", %{conn: conn} do
-      path = conn |> organization_path(:index)
-      conn = conn |> get(path)
+      [organization_1, organization_2] = insert_pair(:organization)
 
-      assert json_response(conn, 200)["data"] == []
+      conn
+      |> request_index
+      |> json_response(200)
+      |> assert_ids_from_response([organization_1.id, organization_2.id])
     end
 
     test "filters resources on index", %{conn: conn} do
-      first_org = insert(:organization, name: "Org A")
-      second_org = insert(:organization, name: "Org B")
-      insert(:organization, name: "Org C")
+      [organization_1, organization_2 | _] = insert_list(3, :organization)
 
-      path = "organizations/?filter[id]=#{first_org.id},#{second_org.id}"
+      path = "organizations/?filter[id]=#{organization_1.id},#{organization_2.id}"
 
       conn
       |> get(path)
       |> json_response(200)
-      |> assert_ids_from_response([first_org.id, second_org.id])
+      |> assert_ids_from_response([organization_1.id, organization_2.id])
     end
   end
 
   describe "show" do
     test "shows chosen resource", %{conn: conn} do
       organization = insert(:organization)
 
-      path = conn |> organization_path(:show, organization)
-      conn = conn |> get(path)
-
-      data = json_response(conn, 200)["data"]
-
-      assert data["id"] == "#{organization.id}"
-      assert data["type"] == "organization"
-      assert data["attributes"]["name"] == organization.name
-      assert data["attributes"]["description"] == organization.description
-      assert data["attributes"]["slug"] == organization.slug
+      conn
+      |> request_show(organization)
+      |> json_response(200)
+      |> Map.get("data")
+      |> assert_result_id(organization.id)
```

Comment:
🤔 When we call `result_id`, we seem to have to retrieve the data hash first, unlike `assert_ids_from_response`, which extracts it for us.  What are you thoughts on updating the `assert_result_id` helper to accept either the data list _or_ the full json response?